src/TestProblem/Hydro/Zeldovich/Init_TestProb_Zeldovich.cpp

```diff
@@ -0,0 +1,537 @@
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Par_Init_ByFunction_Zeldovich
+// Description :  User-specified function to initialize particle attributes
+//
+// Note        :  1. Invoked by Init_GAMER() using the function pointer "Par_Init_ByFunction_Ptr"
+//                   --> This function pointer may be reset by various test problem initializers, in which case
+//                       this funtion will become useless
+//                2. Periodicity should be taken care of in this function
+//                   --> No particles should lie outside the simulation box when the periodic BC is adopted
+//                   --> However, if the non-periodic BC is adopted, particles are allowed to lie outside the box
+//                       (more specifically, outside the "active" region defined by amr->Par->RemoveCell)
+//                       in this function. They will later be removed automatically when calling Par_Aux_InitCheck()
+//                       in Init_GAMER().
+//                3. Particles set by this function are only temporarily stored in this MPI rank
+//                   --> They will later be redistributed when calling Par_FindHomePatch_UniformGrid()
+//                       and LB_Init_LoadBalance()
+//                   --> Therefore, there is no constraint on which particles should be set by this function
+//
+// Parameter   :  NPar_ThisRank : Number of particles to be set by this MPI rank
+//                NPar_AllRank  : Total Number of particles in all MPI ranks
+//                ParMass       : Particle mass     array with the size of NPar_ThisRank
+//                ParPosX/Y/Z   : Particle position array with the size of NPar_ThisRank
+//                ParVelX/Y/Z   : Particle velocity array with the size of NPar_ThisRank
+//                ParTime       : Particle time     array with the size of NPar_ThisRank
+//                ParType       : Particle type     array with the size of NPar_ThisRank
+//                AllAttribute  : Pointer array for all particle attributes
+//                                --> Dimension = [PAR_NATT_TOTAL][NPar_ThisRank]
+//                                --> Use the attribute indices defined in Field.h (e.g., Idx_ParCreTime)
+//                                    to access the data
+//
+// Return      :  ParMass, ParPosX/Y/Z, ParVelX/Y/Z, ParTime, ParType, AllAttribute
+//-------------------------------------------------------------------------------------------------------
+void Par_Init_ByFunction_Zeldovich( const long NPar_ThisRank, const long NPar_AllRank,
+                                  real *ParMass, real *ParPosX, real *ParPosY, real *ParPosZ,
+                                  real *ParVelX, real *ParVelY, real *ParVelZ, real *ParTime,
+                                  real *ParType, real *AllAttribute[PAR_NATT_TOTAL] )
+{
+   #ifdef SUPPORT_GSL
+   if ( Gas_Par_Setup == 2 )
```

Comment:
For simplicity, replace this check by putting `if ( Gas_Par_Setup != 2 )   return;` at the beginning of this function.